lib/quicly.c

```diff
@@ -2029,7 +2039,8 @@ static int client_collected_extensions(ptls_t *tls, ptls_handshake_properties_t
     if ((ret = quicly_decode_transport_parameter_list(&params, needs_cid_auth(conn) || is_retry(conn) ? &original_dcid : NULL,
                                                       needs_cid_auth(conn) ? &initial_scid : &tp_cid_ignore,
                                                       needs_cid_auth(conn) ? is_retry(conn) ? &retry_scid : NULL : &tp_cid_ignore,
-                                                      remote_cid->stateless_reset_token, src, end)) != 0)
+                                                      remote_cid->stateless_reset_token, src, end, recognize_delayed_ack(conn))) !=
```

Comment:
`recognize_delayed_ack` inspects the transport parameter list in `conn` but isn't it going to be set by `quicly_decode_transport_parameter_list`?

Reply:
Okay after looking at the h2o-side change I understood. Initial value is supposed to be set by an application.

Reply:
Correct. There are two sets of transport parameters that we deal with, one being the set of local parameters that is part of `quicly_conn_t`, and the other being the one received from the peer.

Here, we are decoding the values provided by peer, while using the local settings to see if we should respect or ignore the delayed ack extension provided by the peer.